akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -951,6 +951,7 @@ abstract class ShardCoordinator(
   def handleGetShardHome(shard: ShardId): Boolean = {
     if (rebalanceInProgress.contains(shard)) {
       deferGetShardHomeRequest(shard, sender())
+      unstashOneGetShardHomeRequest() // continue unstashing
```

Comment:
it should still unstash for this case, because other requests may be for other shards

Reply:
yes, indeed. And if the next to be unstashed is for a reblalancing shard, then if will it be deferred (added to `rebalancingInProgress`). Basically, moving from one stash to another.